dali/kernels/CMakeLists.txt

```diff
@@ -24,9 +24,40 @@ collect_headers(DALI_INST_HDRS PARENT_SCOPE)
 collect_sources(DALI_KERNEL_SRCS)
 collect_test_sources(DALI_KERNEL_TEST_SRCS)
 
-cuda_add_library(${dali_kernel_lib} STATIC "${DALI_KERNEL_SRCS}")
+cuda_add_library(${dali_kernel_lib} SHARED ${DALI_KERNEL_SRCS})
+set_target_properties(${dali_kernel_lib} PROPERTIES
+    LIBRARY_OUTPUT_DIRECTORY "${DALI_LIBRARY_OUTPUT_DIR}")
+target_link_libraries(${dali_kernel_lib} PUBLIC ${dali_core_lib})
+target_link_libraries(${dali_kernel_lib} PRIVATE "-Wl,--exclude-libs,${exclude_libs}")
+
+set(exports_map_path "${PROJECT_BINARY_DIR}/stage")
+set(lib_exports "libdali_kernels.map")
+configure_file("${DALI_ROOT}/cmake/${lib_exports}.in" "${exports_map_path}/${lib_exports}")
+add_custom_target(${lib_exports} DEPENDS "${exports_map_path}/${lib_exports}")
+add_dependencies(${dali_kernel_lib} ${lib_exports})
+target_link_libraries(${dali_kernel_lib} PRIVATE -Wl,--version-script=${exports_map_path}/${lib_exports})
```

Comment:
This part is repeated for the other libraries, is it worth to do it as a function/macro? At least that map file thing.

Reply:
As we agreed we won't do that because it will obscure some important details